### PR TITLE
Improve tracing/logging on AMQP/MQTT server shutdown

### PR DESCRIPTION
**Pull request by calohmn**
Giving tracing spans created for closing client connections on server shutdown a dedicated operation name.
That makes them stand out from spans created when clients themselves close the connection.

### CODE REVIEW

**Comment by calohmn** [comment]
We *could* also create a span for the server shudown procedure and create the "connection-closing" spans as child spans. But this could create lots and lots of child spans if many clients are connected. So I think the change in this PR is enough for the time being.

File: adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java [hunk 1]
```diff
@@ -241,33 +243,37 @@ private ConnectionLimitManager createConnectionLimitManager() {
 
     @Override
     protected void doStop(final Promise<Void> stopPromise) {
+        if (!stopCalled.compareAndSet(false, true)) {
+            log.trace("stop already called");
+            stopPromise.complete();
```
Comment (sophokles73) [inline adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java:248]: FMPOV this should instead be tied to the outcome of the promise that had initially been used to shut down the adapter, e.g. by using an AtomicReference<Promise<Void>> instead ...

Comment (calohmn) [reply adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: I've amended the commit accordingly.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**